crates/turborepo-lsp/src/lib.rs

```diff
@@ -0,0 +1,853 @@
+                                message: format!(
+                                    "The task `{}` does not exist in the package `{}`.",
+                                    task, package
+                                ),
+                                range: convert_ranges(&rope, object_range),
+                                severity: Some(DiagnosticSeverity::ERROR),
+                                code: Some(NumberOrString::String(
+                                    "turbo:no-such-task".to_string(),
+                                )),
+                                ..Default::default()
+                            });
+                        }
+                        // task exists in a given package, so we're good
+                        (Some(_), Some(_)) => {}
+                        // the task exists and we haven't specified a package, so we're good
+                        (Some(_), None) => {}
+                    }
+                }
+
+                // inputs, outputs
+                globs.extend(
+                    ["inputs", "outputs"]
+                        .iter()
+                        .filter_map(|s| {
+                            property
+                                .value
+                                .as_object()
+                                .and_then(|o| o.get_array(s))
+                                .map(|a| &a.elements)
+                        })
+                        .flatten(),
+                );
+
+                // dependsOn
+                if let Some(array) = property
+                    .value
+                    .as_object()
+                    .and_then(|o| o.get_array("dependsOn"))
+                {
+                    for depends_on in &array.elements {
+                        if let Some(string) = depends_on.as_string_lit() {
+                            if string.value.starts_with('^') {
+                                diagnostics.push(Diagnostic {
+                                    message: format!(
+                                        "The '^' means \"run the `{}` task in the package's \
+                                         depencies before this one\"",
+                                        &string.value[1..],
+                                    ),
+                                    range: convert_ranges(
+                                        &rope,
+                                        collapse_string_range(string.range),
+                                    ),
+                                    severity: Some(DiagnosticSeverity::INFORMATION),
+                                    ..Default::default()
+                                });
+                            }
+                            if string.value.starts_with('$') {
+                                diagnostics.push(Diagnostic {
+                                    message: "The $ syntax is deprecated. Please apply the \
```

Comment:
Do we want to suggest a specific `npx @turbo/codemod` command here?

Reply:
Yes, the diagnostic has a suggested action to apply the codemod. You can right click or use `CMD + .` to open the action list.